rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -91,13 +91,22 @@ void SequentialReader::open(
   // This is necessary for non ROS2 bags (aka ROS1 legacy bags).
   if (metadata_io_->metadata_file_exists(storage_options.uri)) {
     metadata_ = metadata_io_->read_metadata(storage_options.uri);
-    if (metadata_.relative_file_paths.empty()) {
-      ROSBAG2_CPP_LOG_WARN("No file paths were found in metadata.");
-      return;
+    if (metadata_.version < 5) {
```

Comment:
Let's please keep this versioning logic out of the `sequential_reader`.  It's already a bit dangerous, so we should contain it completely in `metadata_io` and `bag_metadata (there may be instances where this is not true right now, but that's not a good reason for us to make the problem worse haha)

see my comment on `BagMetadata` about the redundant fields - maybe this ends up looking like:

```
// resolve_relative_filepaths checks the version (and maybe in the future it's actually a method on BagMetadata to get that logic out of this file)
file_paths_ = details::resolve_relative_paths(storage_options.uri, metadata_);
```